### PR TITLE
Initial fits2isis gtest conversion

### PR DESCRIPTION
**Pull request by AustinSanders**
Converts fits2isis to app and converts existing tests to gtest

## Description
Converts fits2isis/main.cpp into fits2isis.cpp and fits2isis.h.  Adds gtests

The labels for these products are pretty anemic.  There was only a few keywords to test, so I tested all of them in each test.  There is no difference in the values produced by organization bil or bsq, so the tests are largely the same.

## Related Issue
Test conversion

## Motivation and Context
344Mb testing space

## How Has This Been Tested?
3 Gtests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsFits2isis.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  catch (IException &e) {
+    FAIL() << "Unable to ingest fits file: " << e.toString().toStdString().c_str() << std::endl;
+  }
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions Group
+  ASSERT_EQ(cube.sampleCount(), 200);
+  ASSERT_EQ(cube.lineCount(), 25);
+  ASSERT_EQ(cube.bandCount(), 3);
```
Comment (acpaquette) [inline isis/tests/FunctionalTestsFits2isis.cpp]: Using the fits cropper, you should also be able to limit the sample count to 25. Should just be another indexer on the output data array.

File: isis/tests/FunctionalTestsFits2isis.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  ASSERT_NEAR(hist->StandardDeviation(), 0.0245064, .0000001);
+}
+
+TEST(Fits2Isis, Fits2IsisOrganizationBil) {
+  QTemporaryDir prefix;
+  QString dAPP_XML = FileName("$ISISROOT/bin/xml/fits2isis.xml").expanded();
+  QString cubeFileName = prefix.path() + "/fits2isisTEMP.cub";
+  QVector<QString> args = {"from=data/fits2isis/organization.fits",
+                           "to=" + cubeFileName,
+                           "organization=bsq",
```
Comment (acpaquette) [inline isis/tests/FunctionalTestsFits2isis.cpp]: Organization here looks like it should be set to bil.

File: isis/tests/FunctionalTestsFits2isis.cpp [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  catch (IException &e) {
+    FAIL() << "Unable to ingest fits file: " << e.toString().toStdString().c_str() << std::endl;
+  }
+
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+  // Dimensions Group
+  ASSERT_EQ(cube.sampleCount(), 256);
+  ASSERT_EQ(cube.lineCount(), 10);
+  ASSERT_EQ(cube.bandCount(), 2);
```
Comment (acpaquette) [inline isis/tests/FunctionalTestsFits2isis.cpp]: Was this input fits file cropped?

File: isis/tests/FunctionalTestsFits2isis.cpp [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  ASSERT_NEAR(hist->Average(), 0.811103, .000001);
+  ASSERT_NEAR(hist->Sum(), 4055.5169, .0001);
+  ASSERT_EQ(hist->ValidPixels(), 5000);
+  ASSERT_NEAR(hist->StandardDeviation(), 20.1912, .0001);
+}
+
+
+TEST(Fits2Isis, Fits2IsisOrganizationBsq) {
+  QTemporaryDir prefix;
+  QString dAPP_XML = FileName("$ISISROOT/bin/xml/fits2isis.xml").expanded();
```
Comment (acpaquette) [inline isis/tests/FunctionalTestsFits2isis.cpp]: Why is this here? This is already defined in `APP_XML`, so why are we redefining it and using it again?

**Review by acpaquette — COMMENTED** [review]
:+1: Couple things to change and a few questions

File: isis/tests/FunctionalTestsFits2isis.cpp (hunk 4 above)
Comment (AustinSanders) [reply isis/tests/FunctionalTestsFits2isis.cpp]: Using a static APP_XML caused segfaults on jenkins a few months back, but including it as a non-static variable fixed it.  I got a little mixed up here and included both formats.

**Review by tgiroux — DISMISSED** [review]
This looks good to me

**Review by Kelvinrr — CHANGES_REQUESTED** [review]
fits2isis test failure

**Comment by Kelvinrr** [comment]
unfortunately we will have to wait for Jenkins to be back up and I'll start a new build on this 😕

**Review by Kelvinrr — APPROVED** [review]
I'm pretty sure the TGO failures are not related to this PR as I am seeing them in other PRs, something must of broke upstream
